### PR TITLE
Fix Ethereum issue and add easy sweep feature

### LINKED ISSUE #2186: EthereumWatcher stops checking: System.OverflowException, Value was either too large or too small for an Int64.
Hey there, I've edited the `BTCPayServer.Common/Altcoins/Ethereum/BTCPayNetworkProvider.Ethereum.cs` file to add support on a BEP20 token (18 divisibility), everything seem fine with the integration, getting connection etc.

but when i'm paying for an invoice, sometimes it doesn't get confirmed straight away and it throws this error 1-2 times:
```[WRN] Unhandled exception in EthereumWatcher
System.OverflowException: Value was either too large or too small for an Int64.
   at System.Numerics.BigInteger.op_Explicit(BigInteger value)
   at BTCPayServer.Services.Altcoins.Ethereum.Services.EthereumWatcher.GetBalance(EthereumBTCPayNetwork network, BlockParameter blockParameter, String$
   at BTCPayServer.Services.Altcoins.Ethereum.Services.EthereumWatcher.<>c__DisplayClass19_4.<<UpdatePaymentStates>b__9>d.MoveNext() in /root/btcpayse$
--- End of stack trace from previous location where exception was thrown ---
   at BTCPayServer.Services.Altcoins.Ethereum.Services.EthereumWatcher.UpdatePaymentStates(InvoiceEntity[] invoices, CancellationToken cancellationTok$
   at BTCPayServer.Services.Altcoins.Ethereum.Services.EthereumWatcher.UpdateAnyPendingEthLikePaymentAndAddressWatchList(CancellationToken cancellatio$
   at BTCPayServer.Services.Altcoins.Ethereum.Services.EthereumWatcher.ProcessEvent(Object evt, CancellationToken cancellationToken) in /root/btcpayse$
   at BTCPayServer.HostedServices.EventHostedServiceBase.ProcessEvents(CancellationToken cancellationToken) in /root/btcpayserver/BTCPayServer/HostedS$
```

The invoices get created normally,
`Invoice TESTID new event: invoice_created`, i pay for it and no progress is being made.

Seems like EthereumWatcher hangs somewhere and stops updating and checking the invoice

only when i restart btcpayserver these invoices get confirmed at start, and only for small invoices of 1-10 coins.

-tested invoice with 9 coins, paid, restarted btcpayserver and it got confirmed.
-tested invoice with 12 coins or higher, paid, restarted btcpayserver but never got confirmed

i suspect that it has something to do with returning a very big balance amount by divisibility being 18.

Any idea how i might fix this?
Thanks

### PR DESCRIPTION
**Pull request by xpayserver**
hi i fix #2186 and add beta sweep feature 
it group all fund token and eth by wallet and attempt compute most economical path for sweep. when not enough eth it ask send transfer to address in sweep list

### CODE REVIEW

File: BTCPayServer/Security/CookieAuthorizationHandler.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ protected override async Task HandleRequirementAsync(AuthorizationHandlerContext
                     return;
             }
 
-            string storeId = _HttpContext.GetImplicitStoreId();
```
Comment (NicolasDorier) [inline BTCPayServer/Security/CookieAuthorizationHandler.cs:40]: why do you need that?

**Comment by rockstardev** [comment]
https://www.youtube.com/watch?v=YVkUvmDQ3HY

**Comment by rockstardev** [comment]
@xpayserver https://www.youtube.com/watch?v=YVkUvmDQ3HY

File: BTCPayServer/Services/Altcoins/Ethereum/Payments/EthereumLikePaymentData.cs [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
-using Nethereum.Hex.HexTypes;
+using Microsoft.AspNetCore.Http;
 using Nethereum.Web3;
 
 namespace BTCPayServer.Services.Altcoins.Ethereum.Payments
 {
     public class EthereumLikePaymentData : CryptoPaymentData
     {
-        public long Amount { get; set; }
+        public ulong Amount { get; set; }
```
Comment (simonlucalandi) [inline BTCPayServer/Services/Altcoins/Ethereum/Payments/EthereumLikePaymentData.cs:13]: `ulong` is not big enough to map a `uint256` value returned from a web3 call.
For example, in the case of ERC20 with decimals = 18, the balance can be larger than `ulong` max 18,446,744,073,709,551,615
We should use `BigInteger`.

**Comment by pavlenex** [comment]
@xpayserver can you please resolve a conflict and address the comments so we can proceed?

**Comment by Kukks** [comment]
@xpayserver if you want this pr merged for the next release, please fix any concerns and rebase. This has been open for 3 months now and is not our focus.

**Comment by pavlenex** [comment]
Seems like this is abandoned and we can close it. Also #2739 seems to be the successor of this PR.

**Pull request closed without merging**